### DISCUSSION TITLE
How to create a `route` with specific `target`?

### DISCUSSION THREAD

**Discussion by simeon49** [Kong/kong#7439] [Help] [answered]
### Summary
In our usage scenario, we will dynamically generate many temporary intranet service nodes as needed(like http://172.16.2.41:8888/api/, http://172.16.2.42:8888/api/, http://172.16.2.43:8888/api/...), and we need to uniformly generate a dynamic external network access address for them

exp:  
GET  http://www.my-exp.com/dynamically-nodes/41    -> http://172.16.2.41:8888/service/
GET  http://www.my-exp.com/dynamically-nodes/42    -> http://172.16.2.42:8888/service/
GET  http://www.my-exp.com/dynamically-nodes/43    -> http://172.16.2.43:8888/service/
...

config in nginx maybe like:
```yaml
upstream node_41 {
        server 172.16.2.41:8888;
}

upstream node_42 {
        server 172.16.2.42:8888;
}

upstream node_43 {
        server 172.16.2.43:8888;
}
...

server {
        listen 80;
        server_name  www.my-exp.com;
        location /dynamically-nodes/41 {
                proxy_pass http://node_41;
        }
        location /dynamically-nodes/42 {
                proxy_pass http://node_42;
        }
       location /dynamically-nodes/43 {
                proxy_pass http://node_43;
        }
       ....
}
```
my question is how can I create "route" with specific "upstream" or "target"?
or How to use admin API gracefully to achieve my needs?

Kong version: 2.4

**Comment by Tieske** [comment, marked as answer]
Create a `service` (as a destination) and then attach a `route` to match incoming traffic to that destination:
```shell
http -f :8001/services name=node-41 url=http://172.16.2.41:8888/
http -f :8001/services/node-41/routes name=node-41 paths[]=/dynamically-nodes/41
```
